chirp/drivers/tk11.py

```diff
@@ -0,0 +1,2331 @@
+    FM = 0
+    AM = 1
+    LSB = 2
+    USB = 3
+    CW = 4
+
+
+class QTType(enum.Enum):
+    """QT Code type enum"""
+    NONE = 0
+    CTCSS = 1
+    NDCS = 2
+    IDCS = 3
+
+
+class Power(enum.Enum):
+    """Power enum"""
+    LOW = 0
+    MIDDLE = 1
+    HIGH = 2
+
+    @property
+    def level(self):
+        """Get power level"""
+        return {
+            Power.LOW: POWER_LEVELS[0],
+            Power.MIDDLE: POWER_LEVELS[1],
+            Power.HIGH: POWER_LEVELS[2]
+        }[self]
+
+
+def parse_str_from_tk11(data: arrayDataElement, default_str: str = "") -> str:
+    """Parse string from TK11 format to chirp format as str.
+    Remove empty characters."""
+    raw = data.get_raw()
+
+    if not raw or raw[0] == 0xFF:
+        return default_str
+
+    try:
+        return raw.rstrip(b' \x00').decode("ascii")
+    except UnicodeDecodeError:
+        return default_str
+
+
+def parse_str_to_tk11(data: str, length: int = 16) -> list[int]:
+    """Parse string from chirp format to TK11 format.
+    Add empty character until a specific length"""
+    return list(data.ljust(length, "\x00").encode("ascii"))
+
+
+def get_response_code(response) -> int | None:
+    """Get response code or None if the response length is < 2"""
+    if len(response) < 2:
+        return None
+
+    return response[0] | response[1] << 8
+
+
+def content_nor_or(content: bytearray) -> None:
+    """Content nor or.
+    This is a sort of magic obfuscation method.
+    Getting it from TK11 CPS analysis but don't know exactly how it works."""
+    num_array = bytearray([
+        22, 108, 20, 230, 46, 0x91, 13, 64, 33, 53, 213, 64, 19, 3, 233, 128
+    ])
+
+    for i, byte in enumerate(content):
+        content[i] = byte ^ num_array[i % 16]
+
+
+def byte_nor_or(data: bytes, start_index: int, nor_or_len: int) -> bytes:
+    """byte nor or"""
+    content = bytearray(data[start_index:start_index + nor_or_len])
+
+    content_nor_or(content)
+
+    result = bytearray(data)
+    result[start_index:start_index + nor_or_len] = content
+
+    return bytes(result)
+
+
+def encapsulate_message(buf: bytes) -> bytes:
+    """Encapsulate a message with start and end message flags"""
+    length = len(buf)
+    header_bytes = struct.pack(
+        "<BBH",
+        START_MESSAGE_FLAG[0],
+        START_MESSAGE_FLAG[1],
+        length
+    )
+
+    crc_data = crc.crc16_xmodem(buf)
+    ender_bytes = struct.pack("<HBB", crc_data, END_MESSAGE_FLAG[0], END_MESSAGE_FLAG[1])
+
+    data = bytearray()
+    data.extend(header_bytes)
+    data.extend(buf)
+    data.extend(ender_bytes)
+
+    return byte_nor_or(data, 4, length + 2)
+
+
+def create_parameter_read_request(start_address: int, length: int):
+    """Create a parameter read request"""
+    address = 0x080000 + start_address
+
+    message_length = 0xC
+    rsv = (0, 0)
+
+    fmt = '<HHI H BB I'
+
+    return struct.pack(fmt, MessageType.MEMORY_READ_REQUEST, message_length, address, length, rsv[0], rsv[1], MAGIC_CODE)
```

Comment:
Perhaps my fault for giving you the wrong impression... When I said the pylint file is obsolete I meant we don't use pylint for style checking anymore. You still need to pass the flake8 check and the line length requirement is still there and mandatory. That's why you're failing the style target.